common/fitparser.h

```diff
@@ -0,0 +1,100 @@
+/* fitparser.h
+
+Copyright (c) 2022, Nikolaj Schlej. All rights reserved.
+
+This program and the accompanying materials
+are licensed and made available under the terms and conditions of the BSD License
+which accompanies this distribution.  The full text of the license may be found at
+http://opensource.org/licenses/bsd-license.php.
+
+THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+*/
+
+#ifndef FITPARSER_H
+#define FITPARSER_H
+
+#include <vector>
+
+#include "basetypes.h"
+#include "ustring.h"
+#include "ubytearray.h"
+#include "treemodel.h"
+#include "intel_fit.h"
+#include "intel_microcode.h"
+#include "ffsparser.h"
+
+class FfsParser;
+
+#ifdef U_ENABLE_FIT_PARSING_SUPPORT
+class FitParser
+{
+public:
+    // Default constructor and destructor
+    FitParser(TreeModel* treeModel, FfsParser* parser) : model(treeModel), ffsParser(parser),
+        bgAcmFound(false), bgKeyManifestFound(false), bgBootPolicyFound(false) {}
+    ~FitParser() {}
```

Comment:
Do not defined destructor if you do not need it.

Reply:
Copied from other places, needs to be removed everywhere.